veracruz-mcu-client/vc.c

```diff
@@ -0,0 +1,577 @@
+/*
+ * Veracruz client aimed at microcontrollers, built in Zephyr OS
+ *
+ * ## Authors
+ *
+ * The Veracruz Development Team.
+ *
+ * ## Licensing and copyright notice
+ *
+ * See the `LICENSE.markdown` file in the Veracruz root directory for
+ * information on licensing and copyright.
+ *
+ */
+
+#include "vc.h"
+
+#include <random/rand32.h>
+#include <kernel.h>
+
+#include <mbedtls/ssl.h>
+#include <mbedtls/debug.h>
+
+#include "nanopb/pb.h"
+#include "nanopb/pb_encode.h"
+#include "nanopb/pb_decode.h"
+#include "transport_protocol.pb.h"
+
+#include "policy.h"
+#include "xxd.h"
+#include "base64.h"
+#include "http.h"
+#include "clap.h"
+
+
+//// Veracruz session handling ////
+static void mbedtls_debug(void *ctx, int level,
+        const char *file, int line,
+        const char *str) {
+    if (level <= 1) {
+        printf("%s", str);
+    }
+}
+
+static int vc_rawrng(void *p,
+        uint8_t *buf, size_t len) {
+    // TODO use cryptographically secure rng?
+    sys_rand_get(buf, len);
+    return 0;
+}
+
+static ssize_t vc_rawsend(void *p,
+        const uint8_t *buf, size_t len) {
+    vc_t *vc = p;
+
+    if (vc->recv_len != 0) {
+        printf("data left!?!?!?!? %d\n", vc->recv_len);
```

Comment:
Should we handle this error some other way?

Reply:
Ah yes, this is lingering from a debug session. Should probably be an assert.

Reply:
Fixed 👍